plugins/source_metrics.go

```diff
@@ -64,9 +64,7 @@ func (s *SourceMetrics) initWithTables(tables schema.Tables) {
 
 func (s *SourceMetrics) initWithClients(table *schema.Table, clients []schema.ClientMeta) {
 	for _, client := range clients {
-		if _, ok := s.TableClient[table.Name][client.ID()]; !ok {
-			s.TableClient[table.Name][client.ID()] = &TableClientMetrics{}
-		}
+		s.TableClient[table.Name][client.ID()] = &TableClientMetrics{}
```

Comment:
Should this then technically also be clearing the map before it loops over the clients, ie. `s.TableClient[table.Name] = make(map[string]*TableClientMetrics)` so that it's impossible to end up with outdated clients in the map?

Reply:
(We may want to also make a similar fix to `initWithTables` above)

Reply:
Good catches! Have incorporated both into the pr